src/adapters/in_memory.rs

```diff
@@ -56,7 +56,13 @@ impl Database for InMemoryDb {
         let value = guard.get(key)?;
 
         match value.expiry {
```

Comment:
The problem of current and this implementation is that the expired key gets deleted only when it is accessed again.

Reply:
That is also true. 1. I think current implementation will be okay if we additionally implement Cache algorithm. 2. Maybe for clearing cache after it expired, we can use some call back function?